MonoGame.Framework/Platform/Graphics/Texture2D.OpenGL.cs

```diff
@@ -503,5 +506,10 @@ struct SetDataRectState<T>
             public static Action<SetDataRectState<T>> Action =
                 (s) => s.texture.PlatformSetDataBody(s.level, s.arraySlice, s.rect, s.data, s.startIndex, s.elementCount);
         }
+
+        static class GetDataPool<T> where T : struct
+        {
+            public static ArrayPool<T> pool = ArrayPool<T>.Create(int.MaxValue, 1);
```

Comment:
Perhaps add a comment as to why `int.MaxValue` is being used? (I reckon the default 2^20 doesn't work well for our large texture use case?).